### ISSUE TITLE
Improve handling of CS8974

### ISSUE THREAD

**Issue by SupinePandora43** [dotnet/roslyn-analyzers#6816]
### Analyzer

**Diagnostic ID**: CS8974

### Analyzer source

**SDK**: Built-in analyzers in .NET 7 SDK or later

**Version**: SDK 7.0.306

### Describe the bug

Warning  Show up when i call `GCHandle.Alloc(instance.Method)` prior to `Marshal.GetFunctionPointerForDelegate` for application stability.

### Steps To Reproduce

```cs
class TestClass
{
    void Method(){}

    static void Test()
    {
        var t = new TestClass();
        GCHandle.Alloc(t.Method);
    }
}
```

### Expected behavior

No warning

### Actual behavior

`Converting method group 'Method' to non-delegate type 'object'. Did you intend to invoke the method?` 🤓

## Additional context

First, the analyzer should check if delegate can return a boxed (object) type.
Second, ignore calls to `GCHandle.Alloc`.
